### PR TITLE
ChildWorkflow defaults to workflow run timeout specified by parent

### LINKED ISSUE #565: ChildWorkflow inherits WorkflowRunTimeout from parent
Recently we changed definition of timeouts for workflow.  Based on the new definition all timeouts on workflow are now optional.  Here are the new timeouts:
1. WorkflowExecutionTimeout
2. WorkflowRunTimeout
3. WorkflowTaskTimeout

Default for WorkflowExecutionTimeout is 10 years, and default for WorkflowRunTimeout is same as WorkflowExecutionTimeout.
In the case of ChildWorkflow, looks like we inherit WorkflowRunTimeout from parent execution.  This seems counter intuitive in the following case.

Parent is started with WorkflowExecutionTimeout of 1 minute.
```
workflowOptions := client.StartWorkflowOptions{
		ID:                       workflowID,
		TaskList:                 "child-workflow-task-list",
		WorkflowExecutionTimeout: time.Minute,
	}
```

Child Workflow execution is started with WorkflowExecutionTimeout of 30 minutes:
```
cwo := workflow.ChildWorkflowOptions{
		WorkflowID:               childID,
		WorkflowExecutionTimeout: 30 * time.Minute,
	}
```

Now when child is started, workflow execution timeout is 30 minutes but workflow run timeout is 1 minutes:

```
./tctl wf show -w 1 -pdt
  1  2020-05-03T14:03:39-07:00  WorkflowExecutionStarted     {WorkflowType:{Name:SampleChildWorkflow}, ParentWorkflowNamespace:default,
                                                              ParentWorkflowExecution:{WorkflowId:parent-workflow_40416b51-c93a-46ef-a132-904f2f7aad21,
                                                              RunId:ecbfdd56-be57-48be-af24-5d14d63aa402}, ParentInitiatedEventId:5,
                                                              TaskList:{Name:child-workflow-task-list, Kind:Normal},
                                                              Input:{Payloads:[len=2]}, WorkflowExecutionTimeoutSeconds:1800,
                                                              WorkflowRunTimeoutSeconds:60, WorkflowTaskTimeoutSeconds:10,
                                                              Initiator:Decider, OriginalExecutionRunId:5d75373b-867b-454f-ba63-4a52c37dea17,
                                                              FirstExecutionRunId:5d75373b-867b-454f-ba63-4a52c37dea17, Attempt:0,
                                                              WorkflowExecutionExpirationTimestamp:0, FirstDecisionTaskBackoffSeconds:0,
                                                              Header:{Fields:map{}}}
```

### PR DESCRIPTION
**Pull request by samarabbas**
Fixes #565

### CODE REVIEW

File: service/history/commandChecker.go [hunk 1]
```diff
@@ -596,14 +596,22 @@ func (v *commandAttrValidator) validateStartChildExecutionAttributes(
 	}
 	attributes.TaskQueue = taskQueue
 
+	parentSpecifiedExecutionTimeout := attributes.GetWorkflowExecutionTimeoutSeconds()
 	// Inherit workflow timeout from parent workflow execution if not provided on command
```
Comment (alexshtin) [inline service/history/commandChecker.go:600]:
```suggestion
	// Inherit workflow execution timeout from parent workflow execution if not provided on command
```

File: service/history/commandChecker.go [hunk 2]
```diff
@@ -596,14 +596,22 @@ func (v *commandAttrValidator) validateStartChildExecutionAttributes(
...
-	if attributes.GetWorkflowExecutionTimeoutSeconds() <= 0 {
+	if parentSpecifiedExecutionTimeout <= 0 {
 		attributes.WorkflowExecutionTimeoutSeconds = parentInfo.WorkflowExecutionTimeout
 	}
 
 	// Inherit workflow timeout from parent workflow execution if not provided on command
```
Comment (alexshtin) [inline service/history/commandChecker.go:605]:
```suggestion
	// Inherit workflow run timeout from parent workflow execution if not provided on command
```

File: service/history/commandChecker.go [hunk 3]
```diff
@@ -596,14 +596,22 @@ func (v *commandAttrValidator) validateStartChildExecutionAttributes(
 	}
 	attributes.TaskQueue = taskQueue
 
+	parentSpecifiedExecutionTimeout := attributes.GetWorkflowExecutionTimeoutSeconds()
```
Comment (alexshtin) [inline service/history/commandChecker.go:599]: nit: I would call it just `parentExecutionTimeout`.

**Comment by samarabbas** [comment]
@mfateev I have updated PR based on our discussion earlier today to assign timeout default in similar way as parent workflow execution.  Now they don't inherit timeouts from parent.

File: service/history/workflowExecutionUtil.go [hunk 4]
```diff
@@ -222,3 +223,26 @@ func terminateWorkflow(
 	)
 	return err
 }
+
+func getWorkflowExecutionTimeout(namespace string, requestedTimeout int32, serviceConfig *Config) int32 {
```
Comment (mfateev) [inline service/history/workflowExecutionUtil.go:227]: These look like duplicated code with overrideStartWorkflowExecutionRequest.

Comment (samarabbas) [reply service/history/workflowExecutionUtil.go:227]: I refactored `overrideStartWorkflowExecutionRequest` logic to also use these helpers.

**Pull request merged**